Random Users/Resources/Model Controllers/NetworkingController.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  NetworkingController.swift
+//  Random Users
+//
+//  Created by John McCants on 9/11/20.
+//  Copyright © 2020 Erica Sadun. All rights reserved.
+//
+
+import Foundation
+
+class NetworkingController {
+    
+var contacts: Contacts?
+let baseURL = URL(string: "https://randomuser.me/api/?format=json&inc=name,email,phone,picture&results=1000")!
+
+enum URLMethods: String {
+    case get = "GET"
+
+}
+enum NetworkError: Error {
+    case noData
+    case noImageData
+    case decodeFailed
+    case downloadError
+}
+    
+    func getContacts(completion: @escaping (Result<Contacts, NetworkError>) -> Void) {
+        var request = URLRequest(url: baseURL)
+        request.httpMethod = URLMethods.get.rawValue
```

Comment:
You can make your code more reusable and easier to refactor by just having the baseURL be the actually base URL and using query items to construct how you want the URL in the individual networking call.
```suggestion
       let baseURL = URL(string: "https://randomuser.me/api/")! // this would be up above where you have baseURL Currently
       var urlComponents = URLComponents(url: baseURL, resolvingAgainstBaseURL: true)
        
        urlComponents?.queryItems = [
            URLQueryItem(name: "format", value: "json"),
            URLQueryItem(name: "results", value: "1000"),
            URLQueryItem(name: "inc", value: "name,email,phone,picture"),
            URLQueryItem(name: "format", value: "1000")
        ]
        guard let requestURL = urlComponents?.url else {
            completion(someError)
            return
        }
```
or whatever makes sense, like maybe you could have the format json hardcoded, but then the rest set up using query items

Reply:
True that, the way you're saying is more reusable for sure